packages/datepicker/index.ts

```diff
@@ -0,0 +1,428 @@
+      this.navigationDate = newNavigationDate;
+
+      // move keyboard focus once Lit has rendered an updated `data-navigation`
+      // based on this.navigationDate
+      await this.updateComplete;
+      this.selectedCell.focus();
+    }
+  }
+
+  #onCalendarSelect(event: MouseEvent | KeyboardEvent) {
+    // Clicks can hit the `<div>` inside the `<td>`, in
+    // which case we need to get the parentElement to look
+    // up the data-date attribute for the selected value.
+    const isoDate = (event.target as HTMLElement).parentElement.dataset.date || (event.target as HTMLTableCellElement).dataset.date;
+
+    if ('key' in event) {
+      if (event.key === 'Enter' || event.key === ' ') {
+        // Prevents whitespace from being added to the input field
+        event.preventDefault();
+        this.internalValue = this.value = isoDate;
+        this.isCalendarOpen = false;
+        this.#dispatchChangeEvent();
+      }
+    } else {
+      this.internalValue = this.value = isoDate;
+      this.isCalendarOpen = false;
+      this.#dispatchChangeEvent();
+    }
+  }
+
+  connectedCallback(): void {
+    super.connectedCallback();
+
+    this.internalValue = this.value;
+    if (this.value) {
+      const iso = fromISOToDate(this.value);
+      if (iso) {
+        this.navigationDate = iso;
+      }
+    }
+
+    document.addEventListener('mousedown', this.#onClickOutside);
+    document.addEventListener('touchend', this.#onClickOutside);
+    document.addEventListener('focusin', this.#onClickOutside);
+  }
+
+  disconnectedCallback(): void {
+    super.disconnectedCallback();
+
+    document.removeEventListener('mousedown', this.#onClickOutside);
+    document.removeEventListener('touchend', this.#onClickOutside);
+    document.removeEventListener('focusin', this.#onClickOutside);
+  }
+
+  render() {
+    return html`
+      <div class="w-datepicker-wrapper" id="${this.#wrapperId}">
+        <label class="w-datepicker-input-label" for="${this.#inputId}">${this.label}</label>
+        <div class="w-datepicker-input-wrapper">
+          <input
+            id="${this.#inputId}"
+            type="${inputType}"
+            name="${ifDefined(this.name)}"
+            value="${ifDefined(this.value)}"
+            class="w-datepicker-input"
+            @input="${this.#onInput}"
+            @blur="${this.#onInputBlur}"
+            @keydown="${this.#onInputKeyDown}" />
+          <w-button
+            id="${this.#toggleButtonId}"
+            class="w-datepicker-button"
+            aria-label="Open calendar"
+            variant="utility"
+            quiet
+            type="button"
+            @click="${this.#toggleCalendarOpen}">
+            <w-icon-calendar-16></w-icon-calendar-16>
+          </w-button>
+        </div>
+      </div>
+      ${this.isCalendarOpen
+        ? html`
+            <div class="w-dropdown__popover w-dropdown__popover--open">
+              <div
+                aria-roledescription="TODO"
```

Comment:
There are a couple of these text snippets that are part of a `phrases` object in the React version. How do we want to deal with these?